src/Avalonia.Controls/CalendarDatePicker/CalendarDatePicker.cs

```diff
@@ -576,78 +421,57 @@ protected override void OnGotFocus(GotFocusEventArgs e)
                 }
             }
         }
+
+        /// <inheritdoc/>
         protected override void OnLostFocus(RoutedEventArgs e)
         {
             base.OnLostFocus(e);
 
+            _isPressed = false;
+            UpdatePseudoClasses();
+
             SetSelectedDate();
         }
-        
-        private void OnIsDropDownOpenChanged(AvaloniaPropertyChangedEventArgs e)
+
+        /// <inheritdoc/>
+        protected override void OnKeyUp(KeyEventArgs e)
         {
-            var oldValue = (bool)e.OldValue!;
-            var value = (bool)e.NewValue!;
+            var key = e.Key;
 
-            if (_popUp != null && _popUp.Child != null)
+            if ((key == Key.Space || key == Key.Enter) && IsEffectivelyEnabled) // Key.GamepadA is not currently supported
+            {
+                // Since the TextBox is used for direct date entry,
+                // it isn't supported to open the popup/flyout using these keys.
```

Comment:
Just for case, it is possible to handle events with tunnel strategy (alternative of Preview events in uwp). 
Though I don't think it's really needed here for these keys.

Reply:
Yes, I think it's possible to get the actual events. I just don't think it's a good user experience. In this specific case, space is valid for a date and enter means 'commit and update' more than open a flyout. So I didnt enable this functionality and commented why. I think this applies to all input controls.